Detectors/CTP/simulation/src/Digits2Raw.cxx

```diff
@@ -174,46 +177,66 @@ void Digits2Raw::emptyHBFMethod(const header::RDHAny* rdh, std::vector<char>& to
 std::vector<char> Digits2Raw::digits2HBTPayload(const gsl::span<gbtword80_t> digits, uint32_t Npld) const
 {
   std::vector<char> toAdd;
+  int countBytes = 0;
   uint32_t size_gbt = 0;
   gbtword80_t gbtword;
   gbtword80_t gbtsend;
   bool valid;
   for (auto const& dig : digits) {
     valid = makeGBTWord(dig, gbtword, size_gbt, Npld, gbtsend);
+    LOG(info) << Npld << " digit:" << dig << " " << (dig.to_ulong() & 0xfff) << " ";
+    LOG(info) << "gbt  :" << gbtsend << " valid:" << valid;
     if (valid == true) {
       for (uint32_t i = 0; i < NGBT; i += 8) {
         uint32_t w = 0;
         for (uint32_t j = 0; j < 8; j++) {
           w += (1 << j) * gbtsend[i + j];
         }
+        countBytes++;
         char c = w;
         toAdd.push_back(c);
       }
-      // Pad zeros up to 128 bits
-      uint32_t NZeros = (o2::raw::RDHUtils::GBTWord128 * 8 - NGBT) / 8;
-      for (uint32_t i = 0; i < NZeros; i++) {
-        char c = 0;
-        toAdd.push_back(c);
+      if (mPadding) {
+        // Pad zeros up to 128 bits
+        uint32_t NZeros = (o2::raw::RDHUtils::GBTWord128 * 8 - NGBT) / 8;
+        for (uint32_t i = 0; i < NZeros; i++) {
+          char c = 0;
+          toAdd.push_back(c);
+        }
       }
     }
   }
   // add what is left: maybe never left anything - tbc
-  //LOG(info) << size_gbt << " size valid " << valid;
-  //LOG(info) << "gbtword:" << gbtword;
-  //LOG(info) << "gbtsend:" << gbtsend;
+  LOG(info) << size_gbt << " size valid " << valid;
+  LOG(info) << "gbtword:" << gbtword;
+  LOG(info) << "gbtsend:" << gbtsend;
   if (size_gbt > 0) {
-    LOG(debug) << "Adding left over.";
+    LOG(info) << "Adding left over.";
     gbtword80_t gbtsend = gbtword;
     for (uint32_t i = 0; i < NGBT; i += 8) {
       uint32_t w = 0;
       for (uint32_t j = 0; j < 8; j++) {
         w += (1 << j) * gbtsend[i + j];
       }
+      countBytes++;
       char c = w;
       toAdd.push_back(c);
     }
     // Pad zeros up to 128 bits
-    uint32_t NZeros = (o2::raw::RDHUtils::GBTWord128 * 8 - NGBT) / 8;
+    if (mPadding) {
+      uint32_t NZeros = (o2::raw::RDHUtils::GBTWord128 * 8 - NGBT) / 8;
+      for (uint32_t i = 0; i < NZeros; i++) {
+        char c = 0;
+        toAdd.push_back(c);
+      }
+    }
+  }
+  if (mPadding == false) {
+    uint32_t NZeros = 0;
+    if (countBytes % 16) {
+      NZeros = 16 - (countBytes % 16);
+    }
+    LOG(info) << "Adding bytes:" << NZeros;
```

Comment:
is this supposed to guarantee the alignment to 16 bytes? If so, it will be taken care of by the RawFileWriter, which you already notify about the alignment size: https://github.com/AliceO2Group/AliceO2/blob/1523fb14112c639069ccb5e1157cea0922fee567/Detectors/CTP/simulation/src/Digits2Raw.cxx#L52
Note that the alignment filler is `0xff`.

Reply:
I align TF to 128 bits even with no padding. I see your point. I play with it .